src/cohpsk.c

```diff
@@ -962,7 +962,7 @@ void rate_Fs_rx_processing(struct COHPSK *coh, COMP ch_symb[][COHPSK_NC*ND], COM
     int   r, c, i, ch_fdm_frame_index;
     COMP  rx_fdm_frame_bb[COHPSK_M+COHPSK_M/P];
     COMP  rx_baseband[COHPSK_NC*ND][COHPSK_M+COHPSK_M/P];
-    COMP  rx_filt[COHPSK_NC*ND][P+1];
+    COMP  rx_filt[COHPSK_NC*(ND+1)][P+1];
     float env[NT*P], rx_timing;
```

Comment:
why ND+1?  ND is the number of diversity (copies of signal).  It can only be 2.

Reply:
I just set it to that value to make the types "compatible" / same. GCC told me that the accessed buffer `rx_filt` is smaller than the type it is implicitly cast to. If it helps, I can build current master and past the warning here.

Unfortunately, I'm not deep enough into this implementation to understand the semantics of the buffer.